hawkular-alerts-rest-tests/src/test/groovy/org/hawkular/alerts/rest/LifecycleITest.groovy

```diff
@@ -1203,8 +1204,87 @@ class LifecycleITest extends AbstractITestBase {
+        logger.info("Running t10_HWKALERTS234Test")
+        String start = String.valueOf(System.currentTimeMillis());
+
+        // CREATE the trigger
+        def resp = client.get(path: "")
+        assertEquals(200, resp.status)
+
+        Trigger testTrigger = new Trigger("test-HWKALERTS234-trigger", "test-HWKALERTS234-trigger");
+
+        // remove if it exists
+        resp = client.delete(path: "triggers/test-HWKALERTS234-trigger")
+        assert(200 == resp.status || 404 == resp.status)
+
+        testTrigger.setAutoResolve(true);
+        testTrigger.setAutoResolveAlerts(true);
+
+        resp = client.post(path: "triggers", body: testTrigger)
+        assertEquals(200, resp.status)
+
+        // ADD Firing condition
+        MissingCondition firingCond = new MissingCondition("test-HWKALERTS234-trigger",
+                Mode.FIRING, "test-HWKALERTS234-avail", 5000);
+
+        Collection<Condition> conditions = new ArrayList<>(1);
+        conditions.add( firingCond );
+        resp = client.put(path: "triggers/test-HWKALERTS234-trigger/conditions/firing", body: conditions)
+        assertEquals(200, resp.status)
+        assertEquals(1, resp.data.size())
+
+        // ADD AutoResolve condition
+        AvailabilityCondition autoResolveCond = new AvailabilityCondition("test-HWKALERTS234-trigger",
+                Mode.AUTORESOLVE, "test-HWKALERTS234-avail", Operator.UP);
+
+        conditions.clear();
+        conditions.add( autoResolveCond );
+        resp = client.put(path: "triggers/test-HWKALERTS234-trigger/conditions/autoresolve", body: conditions)
+        assertEquals(200, resp.status)
+        assertEquals(1, resp.data.size())
+
+        // ENABLE Trigger
+        testTrigger.setEnabled(true);
+
+        resp = client.put(path: "triggers/test-HWKALERTS234-trigger", body: testTrigger)
+        assertEquals(200, resp.status)
+
+        // FETCH trigger and make sure it's as expected
+        resp = client.get(path: "triggers/test-HWKALERTS234-trigger");
+        assertEquals(200, resp.status)
+        assertEquals("test-HWKALERTS234-trigger", resp.data.name)
+        assertTrue(resp.data.enabled)
+        assertFalse(resp.data.autoDisable);
+        assertFalse(resp.data.autoEnable);
+        assertTrue(resp.data.autoResolve);
+        assertTrue(resp.data.autoResolveAlerts);
+
+        // This tests that one and only one alert is generated, so let it run for up to 10s.
+        for ( int i=0; i < 40; ++i ) {
+            Thread.sleep(500);
+
+            // FETCH recent alerts for trigger, there should be 1
+            resp = client.get(path: "", query: [startTime:start,triggerIds:"test-HWKALERTS234-trigger"] )
+            if ( resp.status == 200 && resp.data.size() >= 2 ) {
+                break;
+            }
+            assertEquals(200, resp.status)
+        }
+        assertEquals(200, resp.status)
+        assertEquals(1, resp.data.size())
+        assertEquals("OPEN", resp.data[0].status)
```

Comment:
@lucasponce OK, so while adding a little more the test I stumbled over a few more tricky things that ended up taking me most of the day to figure out.  I think things are in good shape now, but please take a good look at the changes.